### PR TITLE
Remove uses of at-async in favor of at-spawn

### PR DESCRIPTION
**Pull request by quinnj**
https://github.com/JuliaWeb/HTTP.jl/issues/1038

### CODE REVIEW

**Comment by quinnj** [comment]
Tests currently failing until https://github.com/JuliaServices/ConcurrentUtilities.jl/pull/22 is merged and tagged.

File: src/clientlayers/ExceptionRequest.jl [hunk 1]
```diff
@@ -10,14 +10,15 @@ using ..IOExtras, ..Messages, ..Exceptions
 Throw a `StatusError` if the request returns an error response status.
 """
 function exceptionlayer(handler)
-    return function exceptions(stream; status_exception::Bool=true, logerrors::Bool=false, kw...)
-        res = handler(stream; logerrors=logerrors, kw...)
+    return function exceptions(stream; status_exception::Bool=true, timedout=nothing, logerrors::Bool=false, logtag=nothing, kw...)
```
Comment (Drvi) [inline src/clientlayers/ExceptionRequest.jl:13]: IIUC, `timedout` kwarg is strictly internal thing, the user shouldn't provide their own `TimedOut` struct to a `HTTP.get`? If that is the case, we might want to call it `_timedout` just to make it clearer that this is an internal thing.

File: src/clientlayers/ConnectionRequest.jl [hunk 2]
```diff
@@ -126,8 +127,10 @@ function connectionlayer(handler)
                     break
                 end
             end
-            if logerrors && !(e isa StatusError || e isa TimeoutError)
-                @error "HTTP.ConnectionRequest" exception=(e, catch_backtrace()) method=req.method url=req.url context=req.context
+            root_err = e isa CapturedException ? e.ex : e
+            if logerrors && !(root_err isa StatusError || root_err isa Exceptions.TimeoutError || root_err isa Base.IOError)
```
Comment (Drvi) [inline src/clientlayers/ConnectionRequest.jl:131]: I'm not really sure how to tell if this list of exceptions is right... Would it make sense to only log the error if the exception is not a `CapturedException` (presumably, if it is a CapturedException, we already logged it before we wrapped it)?

**Review by Drvi — APPROVED** [review]
Looks good to me:) I think the `@async -> @spawn` changes are not the most complicated, it's more the switch to OpenSSL and the changes to exception handling which are more difficult to access. But since these are all based on your experiments with large workload, I believe they'll make nice improvements in terms of performance and log deduplication.

File: src/clientlayers/ConnectionRequest.jl (hunk 2 above)
Comment (quinnj) [reply src/clientlayers/ConnectionRequest.jl:131]: Yeah, I like that better

**Pull request merged**